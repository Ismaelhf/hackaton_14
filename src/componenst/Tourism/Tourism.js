import React from 'react';
import {
  StyleSheet,
  View,
  Image,
  TouchableHighlight,
  TouchableOpacity,
  Text,
  ScrollView,
} from 'react-native';

const Tourism = ({data, navigation}) => {
  return (
    <>
      {data.map((item) => (
        <TouchableOpacity
          key={item.id}
          onPress={() => navigation.navigate('Tourism', {item: {item}})}>
          <View>
            <Text>{item.title}</Text>
            <Image
              style={styles.ciudad}
              source={{
                uri: item.image,
              }}
            />
          </View>
        </TouchableOpacity>
      ))}
    </>
  );
};

const styles = StyleSheet.create({
  ciudad: {
    width: 250,
    height: 300,
    marginRight: 10,
  },
});
export default Tourism;
