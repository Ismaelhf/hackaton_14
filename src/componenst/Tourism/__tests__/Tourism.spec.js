// import { shallow, mount, render } from 'enzyme';
import React from 'react';
import {Text, View, TouchableOpacity, Image} from 'react-native';
import {shallow} from 'enzyme';
import Tourism from '../Tourism';
import {tourismData} from '../../../data/tourismData';

const initialProps = {
  data: tourismData,
};

const wrapper = shallow(<Tourism {...initialProps} />);

describe('Tourism component', () => {
  it('should render', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should Tourism element in component', () => {
    expect(wrapper.find(TouchableOpacity).length).toBe(5);
    expect(wrapper.find(View).length).toBe(5);
    expect(wrapper.find(Image).length).toBe(5);
  });
  it('should match props', () => {
    expect(wrapper.props.data).not.toBeNull();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
