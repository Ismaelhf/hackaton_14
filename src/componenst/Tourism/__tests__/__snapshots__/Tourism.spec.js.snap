// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tourism component should match snapshot 1`] = `
<Fragment>
  <ForwardRef
    key="1"
    onPress={[Function]}
  >
    <View>
      <Text>
        Machu pichu
      </Text>
      <Image
        source={
          Object {
            "uri": "https://cdn.getyourguide.com/img/location/5c95efd23efe6.jpeg/148.jpg",
          }
        }
        style={
          Object {
            "height": 300,
            "marginRight": 10,
            "width": 250,
          }
        }
      />
    </View>
  </ForwardRef>
  <ForwardRef
    key="2"
    onPress={[Function]}
  >
    <View>
      <Text>
        Huaraz
      </Text>
      <Image
        source={
          Object {
            "uri": "https://www.peruhop.com/wp-content/uploads/cosas-por-hacer-en-peru-andes-huaraz.jpg",
          }
        }
        style={
          Object {
            "height": 300,
            "marginRight": 10,
            "width": 250,
          }
        }
      />
    </View>
  </ForwardRef>
  <ForwardRef
    key="3"
    onPress={[Function]}
  >
    <View>
      <Text>
        Ica
      </Text>
      <Image
        source={
          Object {
            "uri": "https://www.turiweb.pe/wp-content/uploads/2020/06/huacachina-230620.jpg",
          }
        }
        style={
          Object {
            "height": 300,
            "marginRight": 10,
            "width": 250,
          }
        }
      />
    </View>
  </ForwardRef>
  <ForwardRef
    key="4"
    onPress={[Function]}
  >
    <View>
      <Text>
        7 colores
      </Text>
      <Image
        source={
          Object {
            "uri": "https://www.findlocaltrips.com/Site/ViewSWImage?ImageFileItemId=3996&MaxWidth=350&MaxHeight=275&FixedWidthHeight=True",
          }
        }
        style={
          Object {
            "height": 300,
            "marginRight": 10,
            "width": 250,
          }
        }
      />
    </View>
  </ForwardRef>
  <ForwardRef
    key="5"
    onPress={[Function]}
  >
    <View>
      <Text>
        7 colores
      </Text>
      <Image
        source={
          Object {
            "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTTx-QWMjW5P6B0GuvNOsDzAD5ZPMUU_CN_Pw&usqp=CAU",
          }
        }
        style={
          Object {
            "height": 300,
            "marginRight": 10,
            "width": 250,
          }
        }
      />
    </View>
  </ForwardRef>
</Fragment>
`;
